server/src/main/java/org/apache/iotdb/db/mpp/execution/fragment/FragmentInstanceContext.java

```diff
@@ -72,13 +72,24 @@ public static FragmentInstanceContext createFragmentInstanceContext(
     return instanceContext;
   }
 
+  public static FragmentInstanceContext createFragmentInstanceContextForCompaction(long queryId) {
+    return new FragmentInstanceContext(queryId);
+  }
+
   private FragmentInstanceContext(
       FragmentInstanceId id, FragmentInstanceStateMachine stateMachine) {
     this.id = id;
     this.stateMachine = stateMachine;
     this.executionEndTime.set(END_TIME_INITIAL_VALUE);
   }
 
+  // used for compaction
+  private FragmentInstanceContext(long queryId) {
+    this.queryId = queryId;
```

Comment:
`queryId` will be deprecated. Consider other ways to mock FragmentInstanceContext?